File: dev/LoneWolf/src/com/hoffenkloffen/lonewolf/controllers/RandomNumberTable.java
package com.hoffenkloffen.lonewolf.controllers;

import com.hoffenkloffen.lonewolf.models.RandomNumberResult;

import java.util.Random;

public class RandomNumberTable {

    private Random random;

    public RandomNumberTable() {
        random = new Random();
    }

    public RandomNumberResult getResult()
    {
        return new RandomNumberResult(random.nextInt(10));
    }
}

File: dev/xamarin-old/LoneWolf/LoneWolf/LoneWolf/Domain/RandomNumberTable.cs
using System;
using LoneWolf.Models;

namespace LoneWolf.Domain
{
	public class RandomNumberTable
	{
		private readonly IRandom _random;

		public RandomNumberTable(IRandom random)
		{
			_random = random;
		}

		public RandomNumberResult Next()
		{
			return new RandomNumberResult(_random.Next(10));
		}
	}

	public interface IRandom
	{
		int Next(int maxValue);
	}

	public class RandomFacade : IRandom
	{
		private readonly Random _random;

		public RandomFacade()
		{
			_random = new Random();
		}

		public int Next(int maxValue)
		{
			return _random.Next(maxValue);
		}
	}
}